src/Compilers/CSharp/Portable/Symbols/Source/SourceComplexParameterSymbol.cs

```diff
@@ -200,8 +200,9 @@ protected ConstantValue MakeDefaultExpression(DiagnosticBag diagnostics, Binder
 
             // If we have something like M(double? x = 1) then the expression we'll get is (double?)1, which
             // does not have a constant value. The constant value we want is (double)1.
+            // The default literal conversion is an exception: (double)default would give the wrong value for M(double? x = default).
 
-            if (convertedExpression.ConstantValue == null && convertedExpression.Kind == BoundKind.Conversion)
+            if (convertedExpression.ConstantValue == null && convertedExpression.Kind == BoundKind.Conversion && !convertedExpression.GetConversion().IsNullLiteral)
```

Comment:
you could check `defaultSyntax.Value.Kind`

Reply:
@alrz Good idea, especially considering that checking for `default` using `IsNullLiteral` is confusing. Done.